robot_skills/src/robot_skills/arm/guarded_motion.py

```diff
@@ -0,0 +1,40 @@
+import functools
+import rospy
+import types
+
+from control_msgs.msg import FollowJointTrajectoryGoal, FollowJointTrajectoryAction
+from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
+
+def AddForceSensingArm(arm):
```

Comment:
I am not sure if I agree with this split between the guarded motion and force sensor skills. I agree that they can be different, but I think that each component (a.k.a. arm, base etc) should have their own guarded motion. Only if a guarded motion uses more than 1 component a separate guarded motion skill makes sense in my opinion.

Reply:
This split was made because the guarded motion functionality is a part of the arm and not of the forcesensor. Note that this function should only ever be applied to arms, not to a base (the current implementation is too specific for that). This functionality should be added to any arm that has a force sensor part. This was the cleanest implementation I could think of.

Reply:
I understand where you are coming from: the arm is doing the motion, not the force sensor. But the fact that you need a force sensor in the arm (even placed on a specific (set) of places) to be able to do this motion indicates to me that this is not a 'general' arm guarded motion. For this motion you need both an arm and force_sensor to 'work', so therefore I would reason to put it in the the arm/force_sensor file. That covers all bases right?

Reply:
We could do that but then we are bringing arm specific functionality into a force sensor file. As you said this one depends on both the force sensor and the arm